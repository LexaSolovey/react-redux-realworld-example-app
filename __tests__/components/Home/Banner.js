import React from 'react';
import renderer from 'react-test-renderer';

import Banner from '../../../src/components/Home/Banner';

const token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NTkzNDIsInVzZXJuYW1lIjoieG94YWNoIiwiZXhwIjoxNTY5MDgyNDcwfQ.NuIXH_W2KR6UhXunU8FX3dahFYb8bKhNAoA0ZNWENE4';

describe('>> Banner component', () => {
    it('snapshot null in case when token is exist', () => {
        const tree = renderer.create(<Banner token={token} appName="conduit" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('snapshot without token', () => {
        const tree = renderer.create(<Banner appName="conduit" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders null in case when token is exist', () => {
        const wrapper = renderer.create(<Banner token={token} appName="conduit" />).toJSON();
        expect(wrapper).toEqual(null);
    });
    
})
