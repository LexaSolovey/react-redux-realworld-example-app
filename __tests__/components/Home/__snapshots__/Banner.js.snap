// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> Banner component snapshot null in case when token is exist 1`] = `null`;

exports[`>> Banner component snapshot without token 1`] = `
<div
  className="banner"
>
  <div
    className="container"
  >
    <h1
      className="logo-font"
    >
      conduit
    </h1>
    <p>
      A place to share your knowledge.
    </p>
  </div>
</div>
`;
